ietf/utils/models.py

```diff
@@ -66,153 +68,144 @@ class PromoteMixin(models.Model):
+            [
+                FieldPanel("social_text"),
+                ImageChooserPanel("social_image"),
+                ImageChooserPanel("feed_image"),
+            ],
+            "Social/Meta descriptions",
+        )
     ]
 
     class Meta:
         abstract = True
 
 
-#class NewsAndBlogsMenuItem(Orderable, models.Model):
-#    model = ParentalKey('utils.SecondaryMenu',
-#                        related_name='news_and_blogs_menu_items')
-#    snippet = models.ForeignKey(
-#        'snippets.PrimaryTopic',
-#        related_name='+'
-#    )
-#
-#    panels = [
-#        SnippetChooserPanel('snippet')
-#    ]
-
-
-class ToolsMenuItem(Orderable, models.Model):
-    model = ParentalKey('utils.SecondaryMenu', related_name='tools_menu_items')
+class SubMenuItem(Orderable):
+    parent = ParentalKey("utils.MenuItem", related_name="sub_menu_items")
     page = models.ForeignKey(
-        'wagtailcore.Page',
-        related_name='+',
-        null=True, blank=True,
-        on_delete=models.SET_NULL
+        "wagtailcore.Page",
+        related_name="+",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
     )
     link = models.URLField(blank=True)
-    text = models.CharField(max_length=255, blank=True)
+    text = models.CharField(max_length=40, blank=True)
 
     @property
     def url(self):
-        return self.link or getattr(self.page, 'url', '')
+        return self.link or getattr(self.page, "url", "")
 
     @property
     def title(self):
-        return self.text or getattr(self.page, 'title', '')
+        return self.text or getattr(self.page, "title", "")
 
-    panels = [
-        PageChooserPanel('page'),
-        FieldPanel('link'),
-        FieldPanel('text')
-    ]
+    panels = [PageChooserPanel("page"), FieldPanel("link"), FieldPanel("text")]
 
 
-@register_setting
-class SecondaryMenu(BaseSetting, ClusterableModel):
-    contact_page = models.ForeignKey(
-        'wagtailcore.Page',
-        null=True, blank=True,
-        on_delete=models.SET_NULL,
-        related_name='+'
-    )
-    tools_page = models.ForeignKey(
-        'wagtailcore.Page',
-        null=True, blank=True,
+class MenuItem(ClusterableModel, WagtailOrderable):
+    page = models.ForeignKey(
+        "wagtailcore.Page",
+        null=True,
+        blank=True,
         on_delete=models.SET_NULL,
-        related_name='+'
+        related_name="+",
     )
-
+    text = models.CharField(max_length=40, blank=True)
     panels = [
-#        InlinePanel('news_and_blogs_menu_items', label="News and Blogs Menu"),
-        PageChooserPanel('contact_page'),
-        PageChooserPanel('tools_page'),
-        InlinePanel('tools_menu_items', label="Tools Menu"),
+        FieldPanel("text"),
+        PageChooserPanel("page"),
+        InlinePanel(
+            "sub_menu_items",
+            label="Sub Menu Items",
+        ),
     ]
 
+    @property
+    def is_dropdown(self):
+        return bool(self.sub_menu_items.count())
```

Comment:
You can use `.exists()` here instead -- it returns a bool, and is generally more efficient (though would be trivially different here)